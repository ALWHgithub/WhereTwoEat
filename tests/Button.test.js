import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../screens/components/button'
import StdButtonBlue from '../screens/components/button'
import catButton from '../screens/components/button'

test('renders correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<StdButtonBlue />).toJSON();
    expect(tree).toMatchSnapshot();
  });

test('renders correctly for true', () => {
  const tree = renderer.create(catButton(true,"name")).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly for false', () => {
  const tree = renderer.create(catButton(true,"false")).toJSON();
  expect(tree).toMatchSnapshot();
});