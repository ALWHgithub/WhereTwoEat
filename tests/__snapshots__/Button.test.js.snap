// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "orange",
        "borderRadius": 50,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "marginVertical": 5,
        "paddingHorizontal": 18,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "orange",
        "borderRadius": 50,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "marginVertical": 5,
        "paddingHorizontal": 18,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly for false 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "orange",
        "borderRadius": 50,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "marginVertical": 5,
        "paddingHorizontal": 18,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly for true 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "orange",
        "borderRadius": 50,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "marginVertical": 5,
        "paddingHorizontal": 18,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    />
  </View>
</View>
`;
